src/VisualStudio/Core/Def/Implementation/SemanticClassificationCache/VisualStudioSemanticClassificationCacheService.cs

```diff
@@ -52,18 +52,16 @@ internal class VisualStudioSemanticClassificationCacheService
                 return default;
             }
 
-            var classifiedSpans = await client.RunRemoteAsync<SerializableClassifiedSpans>(
-                WellKnownServiceHubService.CodeAnalysis,
-                nameof(IRemoteSemanticClassificationCacheService.GetCachedSemanticClassificationsAsync),
-                solution: null,
-                arguments: new object[] { documentKey.Dehydrate(), textSpan, checksum },
+            var classifiedSpans = await client.TryInvokeAsync<IRemoteSemanticClassificationCacheService, SerializableClassifiedSpans?>(
+                (service, cancellationToken) => service.GetCachedSemanticClassificationsAsync(documentKey.Dehydrate(), textSpan, checksum, cancellationToken),
```

Comment:
when your work goes in, i'm happy to get rid of the Dehydrate/REhyrdrate code to use messagepack.

Reply:
Yeah, some clean up will be possible.